### DISCUSSION TITLE
Question about "linkWithCredential" in auth

### DISCUSSION THREAD

**Discussion by cqwcent** [invertase/react-native-firebase#4910] [Q&A] [answered]
Hi guys,
I am testing email login with social login for my app. However I found a use case that is not behaving as expected.

Preparation steps:
1. Create an email login, say "abc@gmail.com"
2. Link FB to this account
3. Try to bind FBagain, and get "provider-already-linked" (expected)

So far, everything is working fine. With either way to login, I am able to retrieve the same user information. Now I have an account, let's call it ABC_account, with both 2 ways to login to. This is nice.

However, the following use case is not.

1. Create another email login, "xyz@gmail.com"
2. **On Android: Link the same FB account to it. "auth" subscriber detect ABC_account, and automatically changed the current user to ABC_ACCOUNT.**
3. On Android: Link the same FB account to it. got error saying credential in use. This is expected.

Expecting:
The same error message on Android device.

"@react-native-firebase/app": "^10.6.4",
"@react-native-firebase/auth": "^10.6.4",
"react-native": "0.63.4",

(Sorry, I'm not able to find template for question/issues, please let me know and I'll update it.)

**Comment by mikehardy** [comment, marked as answer]
The issue template is for issues, which this is - you opened it as a discussion though, so the template does not really appear
Interesting case, are you able to detect that prior to attempting to link the already-linked FB account that the FB account as a provider is used? Just curious - I think in my app I check "availability" prior to attempting to the link with getProvidersForEmail or similar

**Reply by cqwcent** [reply]
> The issue template is for issues, which this is - you opened it as a discussion though, so the template does not really appear
> Interesting case, are you able to detect that prior to attempting to link the already-linked FB account that the FB account as a provider is used? Just curious - I think in my app I check "availability" prior to attempting to the link with getProvidersForEmail or similar

Thanks, I will move it to issue tab. For your question, I think you mean by "fetchSignInMethodsForEmail", which returns a list of providers for given email address. And yes, but for the "XYZ" account as I mentioned, this method only returns "password" as a provider. And for the "ABC" account, this method returns "password" and "facebook.com".
